test_suite/tests/test_annotations.rs

```diff
@@ -1815,6 +1815,32 @@ fn test_flatten_unit() {
     );
 }
 
+#[test]
+fn test_flatten_unit_struct() {
```

Comment:
It would be great if new flatten tests were added to the `flatten` module in the end of this file. Just my opinion